trin-history/src/jsonrpc.rs

```diff
@@ -19,6 +24,23 @@ impl HistoryRequestHandler {
                         self.network.read().await.overlay.data_radius.to_string(),
                     )));
                 }
+                HistoryEndpoint::Ping => {
```

Comment:
I'm curious if anybody has some insight into why the History network requires a `RwLock` here? I noticed, the state network doesn't. I also tried removing this `RwLock`, but after a fair amount of digging, wasn't able to resolve it. I'm happy to pick that thread back up in another pr, it seems like we've adopted the strategy of the fewer locks the better. But curious to hear any other thoughts first

Reply:
The state network had one until 048c1eca5f16597b5b0502c420ee10166ba72925 in #178 -- that commit might help. Though it's probably just big enough that it's worth being in its own PR.

Reply:
:+1: tracking it in an issue for now